app/scripts/lib/multichain-api/provider-request.js

```diff
@@ -18,38 +11,45 @@ export async function providerRequestHandler(
   end,
   hooks,
 ) {
-  const [scope, wrappedRequest] = Array.isArray(request.params)
-    ? request.params
-    : paramsToArray(request.params);
+  const { scope, request: wrappedRequest } = request.params;
 
-  if (!hooks.hasPermission(request.origin, Caip25EndowmentPermissionName)) {
+  const caveat = hooks.getCaveat(
+    request.origin,
+    Caip25EndowmentPermissionName,
+    Caip25CaveatType,
+  );
+  if (!caveat) {
     return end(new Error('missing CAIP-25 endowment'));
   }
 
-  const chainId = scope.split(':')[1];
+  // TODO: consider case when scope is defined in requireScopes and optionalScopes
+  const scopeObject =
+    caveat.value.requiredScopes[scope] || caveat.value.optionalScopes[scope];
+
+  if (!scopeObject) {
+    return end(new Error('unauthorized (scopeObject missing)'));
+  }
 
-  if (!chainId) {
-    return end(new Error('missing chainId'));
+  if (!scopeObject.methods.includes(wrappedRequest.method)) {
+    return end(new Error('unauthorized (method missing in scopeObject)'));
```

Comment:
```suggestion
    return end(new Error('unauthorized (method not authorized)'));
```

Reply:
we were going to come back and decide on proper error codes and messages at a later point haha